README.md

```diff
@@ -1,2 +1,243 @@
+- [ ] SIP 006: Clarity Execution Cost Assessment (Q1 2020)
+
+Stacks improvement proposals (SIP) are documented design proposals aimed at improving the Stacks blockchain. They should contain concise technical specifications of features or standards and the rationale behind it. SIPs are intended to be the primary mechanism of proposing new features, for collecting of community input on an issue and for documenting design decisions.
+
+See [SIP 000](https://github.com/blockstack/blockstack-core/blob/develop/sip/sip-000-stacks-improvement-proposal-process.md) for more details.
+
+### Testnet versions
+
+- [x] **Local Testnet** is a developer local setup, mono-node, assembling SIP 001, SIP 002, SIP 004 and SIP 005. With this version, developers can run Stack 2.0 on their development machines, but also write, execute and tests smart contracts. See the instructions below for more details.
+
+- [ ] **Open Testnet** is the incoming version of our testnet, that we're intending to ship in Q1 2020. This testnet will ship with SIP 003 and be an open membership network, where participants will be able to observe and experience the consensus running at scale.
+
+- [ ] **Mainet** is the fully functional version, that we're intending to ship in Q2 2020.
+
+## Getting started
+
+### Download and build blockstack-core
+
+The first step is to ensure that you have Rust and the support software installed.
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+From there, you can clone this repository:
+
+```bash
+git clone https://github.com/blockstack/blockstack-core.git
+
+cd blockstack-core
+```
+
+Then build the project:ca
+
+```bash
+cargo build
+```
+
+And run the tests:
+
+```bash
+cargo test testnet  -- --test-threads=1
+```
+
+### Encode and sign transactions
+
+Let's start by generating a keypair, that will be used for signing the upcoming transactions:
+
+```bash
+cargo run --bin blockstack-cli generate-sk --testnet
+
+# Output
+# {
+#  secretKey: "b8d99fd45da58038d630d9855d3ca2466e8e0f89d3894c4724f0efc9ff4b51f001",
+#  publicKey: "02781d2d3a545afdb7f6013a8241b9e400475397516a0d0f76863c6742210539b5",
+#  stacksAddress: "ST2ZRX0K27GW0SP3GJCEMHD95TQGJMKB7G9Y0X1MH"
+# }
+```
+
+We are considering the following simple contract `kv-store`, that you can save to `./kv-store.clar`:
+
+```scheme
+(define-map store ((key (buff 32))) ((value (buff 32))))
+
+(define-public (get-value (key (buff 32)))
+    (match (map-get? store ((key key)))
+        entry (ok (get value entry))
+        (err 0)))
+
+(define-public (set-value (key (buff 32)) (value (buff 32)))
+    (begin
+        (map-set store ((key key)) ((value value)))
+        (ok 'true)))
+```
+
+We want to publish this contract on chain, and issuing some transactions interacting with it, setting some keys and getting some values, so we can observe read and writes.
```

Comment:
We want to publish this contract on chain, then issue some transactions that interact with it by setting some keys and getting some values, so we can observe read and writes.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1